WordPress/wordcamp.org
# Misc: Switch from select2 to selectWoo

[SelectWoo](https://github.com/woocommerce/selectWoo) is a drop-in replacement for select2 which fixes most accessibility issues with select2. There should be no visual changes with this PR, just improved keyboard & screen reader accessibility.

See #458, if we need to add select2-style functionality to new projects, we should update to the more-accessible version.

Another thing to note, SelectWoo appears semi-abandoned, and select2 has some motion again including accessibility fixes, so we might be able to swap back at some point.

### How to test the changes in this Pull Request:

1. Spot check the places that use select2, they should still work
   - WordCamp post type: used on selects in the edit screen, in "Information" metabox; in mentorship dropdown
   - Budgets: used for currency select, sponsor invoice & payment request fields
   - Reports: used on status selects for meetup & wordcamp status reports, wordcamp count report

Try testing with keyboard too, it should work better now. VoiceOver testing was still pretty spotty for me, but at least it allows me to search for items.
